### ISSUE TITLE
Save disabled/enabled action status (also) in the database

### ISSUE THREAD

**Issue by pietervogelaar** [StackStorm/st2#3579]
I have a high available StackStorm setup that is managed with Puppet. For several batch jobs I have an action with cron timer rule. Sometimes it's desirable that a job can temporarily be quickly disabled. But at the moment it's part of the rule/action definition, so a change has to go through Bitbucket, git-flow, Puppet Master.

It would be nice if I could toggle the disabled/enabled status with the st2web GUI that would save the status into the (MongoDB) database. With maybe a reset status option to go back to the status that is configured in the rule/action definition.

**Comment by LindsayHill** [comment]
You can enable/disable a rule by checking/un-checking the "enabled" checkbox in the rule definition in the st2web UI. See this screenshot

<img width="833" alt="rule_disable" src="https://user-images.githubusercontent.com/3607046/28343919-ffb06f8a-6c73-11e7-81b2-44c1987b88b1.png">

Changing that will update the value in MongoDB, the same as using `st2 rule disable`. 

Does that not do what you want?

**Comment by pietervogelaar** [comment]
Clicking that checkbox does absolutely nothing, no AJAX request is fired. The rules are put on all the StackStorm nodes with Puppet. Could it be the case that clicking the checkbox only works if you manually add a rule through the UI?

**Comment by LindsayHill** [comment]
@pietervogelaar click on "Edit", then check/un-check "enabled"

I just tested this with my system. Before doing anything, I had this:

```sh
vagrant@bwcvagrant:~$ st2 rule list -p default
+-----------------------------------+---------+-----------------------------------+---------+
| ref                               | pack    | description                       | enabled |
+-----------------------------------+---------+-----------------------------------+---------+
| default.pool_decommission_rule_wi | default | rule to trigger pool              | True    |
| th_webhook                        |         | decommission.                     |         |
| default.test_rule                 | default |                                   | False   |
+-----------------------------------+---------+-----------------------------------+---------+
vagrant@bwcvagrant:~$
```

I then went to the Rules tab, and found that rule. I clicked Edit, then checked the "enabled" box. That resulted in this:

```sh
vagrant@bwcvagrant:~$ st2 rule list -p default
+-----------------------------------+---------+-----------------------------------+---------+
| ref                               | pack    | description                       | enabled |
+-----------------------------------+---------+-----------------------------------+---------+
| default.pool_decommission_rule_wi | default | rule to trigger pool              | True    |
| th_webhook                        |         | decommission.                     |         |
| default.test_rule                 | default |                                   | True    |
+-----------------------------------+---------+-----------------------------------+---------+
vagrant@bwcvagrant:~$```

There's nothing magic about how the rules got onto the system. Once they're in the DB, then the UI is acting on rules in the DB. It's not referring to the filesystem again until ST2 is told to reload that content.

**Comment by pietervogelaar** [comment]
Ah you're right, if I click edit first and then toggle the checkbox everything works great! But I still think the checkbox should not be clickable when not in edit mode. It's counter intuitive at the moment.

**Comment by LindsayHill** [comment]
I'm not sure it is clickable if it's not in Edit mode - it is the same as clicking anywhere on a web page - nothing happens. There's no indication of the checkbox being enabled or disabled, if I'm not in Edit mode. I'm not sure what else it should do?

**Comment by LindsayHill** [comment]
Or are you seeing the check box get enabled/disabled when not in Edit mode? I've got a feeling there might have been a recent behavior change here

**Comment by pietervogelaar** [comment]
Yes exactly that. I see it change to enabled or disabled when not in edit mode.

**Comment by LindsayHill** [comment]
See https://github.com/StackStorm/st2web/pull/361 - this was resolved in 2.3.0.
